src/licensure.rs

```diff
@@ -57,6 +59,17 @@ impl Licensure {
                 }
             }
 
+            let shebang_re: Regex = Regex::new(r"^#!.*\n").unwrap();
+            let full_content: String = content.clone();
```

Comment:
It's not obvious to me why this clone is required. If you do then please comment here but I will pull this down and try playing with it myself this afternoon

Reply:
I originally had:

```rust
            let shebang_re: Regex = Regex::new(r"^#!.*\n").unwrap();
            let shebang_match_opt = shebang_re.find(&content);
            let shebang_opt = match shebang_match_opt {
                Some(shebang_match) => {
                    content = content.split_off(shebang_match.end());
                    Option::Some(shebang_match.as_str())
                }
                None => Option::None
            };
```

Which resulted in the error:

```
error[E0502]: cannot borrow `content` as mutable because it is also borrowed as immutable
  --> src/licensure.rs:66:31
   |
63 |             let shebang_match_opt = shebang_re.find(&content);
   |                                                     -------- immutable borrow occurs here
...
66 |                     content = content.split_off(shebang_match.end());
   |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ mutable borrow occurs here
67 |                     Option::Some(shebang_match.as_str())
   |                                  ------------- immutable borrow later used here

```

I'll freely admit I'm still learning my way around Rust, so happy for to apply a better approach.